mitmproxy/addons/maplocal.py

```diff
@@ -133,11 +133,15 @@ def request(self, flow: http.HTTPFlow) -> None:
                 if mimetype:
                     headers = {"Content-Type": mimetype}
                 if local_file:
-                    flow.response = http.HTTPResponse.make(
-                        200,
-                        local_file.read_bytes(),
-                        headers
-                    )
+                    try:
+                        flow.response = http.HTTPResponse.make(
+                            200,
+                            local_file.read_bytes(),
+                            headers
+                        )
+                    except IOError as e:
+                        ctx.log.warn(f"Could not read file: {e}")
```

Comment:
Hey @mhils, any idea how to test this code. Above we perform `is_dir()` and `is_file()`, with no chance to delete the file in between. Can we trigger the `IOError` somehow?

Reply:
How about using pytest's monkeypatch functionality?

Reply:
Great, thanks.